File: packages/c#/Model/Robot.cs
namespace Com.CodeGame.CodeBall2018.DevKit.CSharpCgdk.Model
{
    public sealed class Robot : ISphere
    {
        public int id { get; set; }
        public int player_id { get; set; }
        public bool is_teammate { get; set; }
        public double x { get; set; }
        public double y { get; set; }
        public double z { get; set; }
        public double velocity_x { get; set; }
        public double velocity_y { get; set; }
        public double velocity_z { get; set; }
        public double radius { get; set; }
        public double nitro_amount { get; set; }
        public bool touch { get; set; }
        public double? touch_normal_x { get; set; }
        public double? touch_normal_y { get; set; }
        public double? touch_normal_z { get; set; }
    }
}

File: packages/rust/src/model/robot.rs
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Robot {
    pub id: i32,
    pub player_id: i32,
    pub is_teammate: bool,
    pub x: f64,
    pub y: f64,
    pub z: f64,
    pub velocity_x: f64,
    pub velocity_y: f64,
    pub velocity_z: f64,
    pub radius: f64,
    pub nitro_amount: f64,
    pub touch: bool,
    pub touch_normal_x: Option<f64>,
    pub touch_normal_y: Option<f64>,
    pub touch_normal_z: Option<f64>,
}
